strawberry_django/type.py

```diff
@@ -174,13 +174,15 @@ def process_type(
     # update annotations and fields
     cls.__annotations__ = cls_annotations = {}
     for field in fields:
-        annotation = (
-            field.type
-            if field.type_annotation is None
-            else field.type_annotation.annotation
-        )
-        if annotation is None:
-            annotation = StrawberryAnnotation(strawberry.auto)
+        annotation = None
+
+        if field.type_annotation and field.type_annotation.annotation:
+            annotation = field.type_annotation.annotation
+        elif field.base_resolver and field.base_resolver.type_annotation:
+            annotation = field.base_resolver.type_annotation.annotation
```

Comment:
here we try to get the annotation from the base resolver if we didn't get one before

Reply:
@patrick91 Why aren't you using `field.type`?

Reply:
field.type tries to use .resolve and it fails because it is too early: 
<img width="656" alt="CleanShot 2022-11-04 at 18 35 19@2x" src="https://user-images.githubusercontent.com/667029/200050252-733d4039-b460-4b26-863c-9f5b2893e7ce.png">

Reply:
Hmm ok. I think we need a `resolved_type` method (or `unresolved_type`) which encapsulates this logic otherwise we're just spreading it out. It will break if the `graphql_type` PR lands for example.

Reply:
yes totally :D I think we also need find a way to consolidate the logic for getting the field and types with the one we use for the pydantic extension

Reply:
I agree with @jkimbo that we really need a `resolve_type` method =P